physx/src/foundation.rs

```diff
@@ -84,14 +84,31 @@ pub trait Foundation: Class<physx_sys::PxFoundation> + Sized {
     /// Returns `None` if `phys_PxCreateFoundation` returns a null pointer.
     fn new(allocator: Self::Allocator) -> Option<Owner<Self>> {
         unsafe {
-            Owner::from_raw(phys_PxCreateFoundation(
-                crate::physics::PX_PHYSICS_VERSION,
-                allocator.into_px(),
+            Self::with_allocator_error_callback(
+                allocator,
                 get_default_error_callback() as *mut PxErrorCallback,
```

Comment:
If we want to handle deallocattion of the callback later. How will we tell the difference between this singleton, and a pointer created by `create_error_callback`?